submission/submission-webapp/src/itest/kotlin/ac/uk/ebi/biostd/itest/test/submission/refresh/SubmissionRefreshApiTest.kt

```diff
@@ -99,48 +105,47 @@ internal class SubmissionApiTest(private val tempFolder: TemporaryFolder) : Base
             }
 
             webClient.submitSingle(submission, TSV, listOf(refreshFile))
-            val dbSubmission = getSubmissionDb()
-            assertSubmission(
-                submission = getSubmissionDb(),
+            assertExtSubmission(
+                extSubmission = getExtSubmission(),
                 title = SUBTITLE,
                 releaseTime = releaseDate,
                 accessTags = listOf(PUBLIC_ACCESS_TAG.value),
                 attributes = listOf(ATTR_NAME to ATTR_VALUE))
 
-            updateSubmission(dbSubmission)
+            updateSubmission(getSubmissionDb())
             webClient.refreshSubmission(ACC_NO)
 
-            val stored = getSubmissionDb()
-            assertSubmission(
-                submission = stored,
+            val stored = getExtSubmission()
+            assertExtSubmission(
+                extSubmission = stored,
                 title = NEW_SUBTITLE,
                 releaseTime = newReleaseDate,
                 accessTags = emptyList(),
                 attributes = listOf(ATTR_NAME to NEW_ATTR_VALUE))
         }
 
-        private fun assertSubmission(
-            submission: DbSubmission,
+        private fun assertExtSubmission(
+            extSubmission: ExtSubmission,
             title: String,
             releaseTime: OffsetDateTime,
             accessTags: List<String>,
             attributes: List<Pair<String, String>>
         ) {
-            assertThat(submission.title).isEqualTo(title)
-            assertThat(submission.releaseTime).isEqualTo(releaseTime)
-            assertThat(submission.rootPath).isEqualTo(ROOT_PATH)
-            assertThat(submission.accessTags.map { it.name }).containsExactlyElementsOf(accessTags)
-            assertThat(submission.attributes.map { it.name to it.value }).containsExactlyElementsOf(attributes)
+            assertThat(extSubmission.title).isEqualTo(title)
+            assertThat(extSubmission.releaseTime).isEqualTo(releaseTime)
+            assertThat(extSubmission.rootPath).isEqualTo(ROOT_PATH)
+            assertThat(extSubmission.accessTags.map { it.name }).containsExactlyElementsOf(accessTags)
+            assertThat(extSubmission.attributes.map { it.name to it.value }).containsExactlyElementsOf(attributes)
 
-            assertThat(submission.rootSection.type).isEqualTo("Study")
-            assertThat(submission.rootSection.files).hasSize(2)
+            assertThat(extSubmission.section.type).isEqualTo("Study")
+            assertThat(extSubmission.section.files).hasSize(2)
 
-            assertFile(submission.rootSection.files.first(), "regular")
-            assertFile(submission.rootSection.files.last(), "duplicated")
+            extSubmission.section.files.first().ifLeft { assertFile(it, "regular") }
```

Comment:
I think this is incorrect as if it is a right test will not fail. We need to ensure is a left AND  is the expected file (same for the next file)

Reply:
Got it